File: DownloadFiles/TGNet/Chapter08/VB/Exercise8_3/MaxOrdersTestSQLServer/MaxOrdersTest/Form1.vb
Imports System.Data.SqlClient
Public Class Form1

    Private Sub cmdADO_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdADO.Click
        Dim cn As New SqlConnection("Data Source=localhost;Database=Northwind;Trusted_Connection=Yes")
        Dim dr As SqlDataReader
        Dim iRet As Int16
        Dim iCount As Int16 = 0

        cn.Open()
        Dim cmd As New SqlCommand
        cmd.CommandText = _
        "SELECT Orderid, Orderdate FROM Orders" & _
          "WHERE orderdate=(SELECT MAX(orderdate) FROM orders)"
        cmd.Connection = cn
        dr = cmd.ExecuteReader
        If dr.HasRows Then
            Do While dr.Read
                lstResults.Items.Add(dr.Item("OrderID").ToString + " " + _
                    dr.Item("Orderdate").ToString)
                iCount += 1
            Loop
            MessageBox.Show(iCount.ToString & " rows qualified")
        Else
            MessageBox.Show("no rows returned")
        End If
        dr.Close()
        cn.Close()


        'Const iEXPECTED As Integer = 91 'set number of expected items
        cn.Open()
        iRet = cmd.ExecuteScalar
    End Sub
End Class

File: DownloadFiles/TGNet/Chapter08/C#/Exercise8-3/MaxOrdersTestSQLServer/MaxOrdersTest/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Diagnostics;
using System.Data.SqlClient;

namespace MaxOrdersTest
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void cmdADO_Click(object sender, EventArgs e)
        {
            SqlConnection cn = new SqlConnection("Data Source=localhost;Database=Northwind;Trusted_Connection=Yes");
            SqlDataReader dr;
            Int16 iCount = 0;
            cn.Open();
            SqlCommand cmd = new SqlCommand();
            cmd.CommandText = "SELECT Orderid, Orderdate FROM Orders " + 
                "WHERE orderdate = (SELECT MAX(orderdate) FROM orders)";
            cmd.Connection = cn;
            dr = cmd.ExecuteReader();
            if (dr.HasRows)
            {
                while (dr.Read())
                {
                    lstResults.Items.Add
                        (dr.GetInt32 (dr.GetOrdinal("OrderID"))
                           + " " + dr.GetDateTime(dr.GetOrdinal ("Orderdate")));
                    iCount += 1;
                }
                MessageBox.Show(iCount + " rows qualified");
            }
            else
            {              MessageBox.Show("no rows returned");
            }
            dr.Close();
            cn.Close();
        }
    }
}